mlflow/projects/utils.py

```diff
@@ -86,7 +86,7 @@ def _is_file_uri(uri):
 
 def _is_local_uri(uri):
     """Returns True if the passed-in URI should be interpreted as a path on the local filesystem."""
-    return not _GIT_URI_REGEX.match(uri)
+    return not _GIT_URI_REGEX.match(uri) or (os.name == 'nt' and re.match('^[a-zA-Z]:\\\\.*', uri))
```

Comment:
Can we call `is_valid_windows_path` here?

Reply:
Seems surprisingly hard to do this correctly...see https://stackoverflow.com/a/34102855/12111347

At least in our case, it seems we always use `_is_local_uri` to check whether a string is a file URI (`file:///Users/apurva/...`) or a local path (`/Users/apurva/...`) _that actually exists_ (lmk if I missed something!). Given that we expect the file to exist in these cases, maybe we can actually remove this function and update usages of it to check `os.path.isfile(uri) or os.path.isdir(uri)` etc. The one downside is that if you pass a nonexistent local path (e.g. `/Users/not-apurva/...`) we may attempt to treat it as a remote URI & git clone it etc, resulting in an odd error message - we may just want to audit what error messages show up and ensure they're clear.